inject/src/main/java/io/micronaut/context/env/PropertySourcePropertyResolver.java

```diff
@@ -885,6 +892,36 @@ private void fill(List list, Integer toIndex, Object value) {
         }
     }
 
+    private int getNextIntegerInRange(String range) {
+        String[] tokens = range.split(",");
+        int lowerBound = Integer.parseInt(tokens[0]);
+        if (tokens.length == 1) {
+            return lowerBound >= 0 ? 1 : -1  * (random.nextInt(Math.abs(lowerBound)));
+        }
+        int upperBound = Integer.parseInt(tokens[1]);
+        return lowerBound + (int) (Math.random() * (upperBound - lowerBound));
+    }
+
+    private long getNextLongInRange(String range) {
+        String[] tokens = range.split(",");
+        long lowerBound = Long.parseLong(tokens[0]);
+        if (tokens.length == 1) {
+            return (long) (Math.random() * (lowerBound));
+        }
+        long upperBound = Long.parseLong(tokens[1]);
+        return lowerBound + (long) (Math.random() * (upperBound - lowerBound));
+    }
+
+    private float getNextFloatInRange(String range) {
+        String[] tokens = range.split(",");
+        float lowerBound = Float.parseFloat(tokens[0]);
+        if (tokens.length == 1) {
+            return (float) (Math.random() * (lowerBound));
+        }
+        float upperBound = Float.parseFloat(tokens[1]);
+        return lowerBound + (float) (Math.random() * (upperBound - lowerBound));
+    }
+
```

Comment:
I think the only thing missing here is better error handling. If the user doesn't use the feature correctly they would get `NumberFormatException`, etc... however that wouldn't be a very helpful error message. This logic should catch any exceptions thrown by `parse` methods and report a more helpful message to indicate to the user what they did wrong.

Reply:
@jameskleeh is the last commit I pushed is something you want?

Reply:
@utsav0209 I would instead throw a ValueException with the original passed in as the cause

Reply:
Okay understood, will nake the change and should I keep the Log traces?

Reply:
The exception will be logged so no need for additional logs